src/protocols/secure_channel/CASESession.cpp

```diff
@@ -283,7 +296,7 @@ CHIP_ERROR CASESession::DeriveSecureSession(const uint8_t * info, size_t info_le
 
 CHIP_ERROR CASESession::SendSigmaR1()
 {
-    uint16_t data_len = static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) +
+    uint16_t data_len = static_cast<uint16_t>(kSigmaParamRandomNumberSize + sizeof(uint16_t) + sizeof(uint16_t) +
```

Comment:
could we extract the sigma1 encoding into some separate file with unit tests? These sizeof calculations don't seem obvious to double-check

Reply:
The message construction and parsing will be changed to TLV formatting. So this code will get reworked. I agree that we should extract out the logic to help unit test it.